spotbugs/src/main/java/edu/umd/cs/findbugs/SourceLineAnnotation.java

```diff
@@ -944,6 +944,36 @@ public String getSourcePath() {
         return sourcePath;
     }
 
+    public String getRealSourcePath() {
+        if (isSourceFileKnown()) {
+            SourceFinder sourceFinder = getSourceFinder();
+            if (sourceFinder != null)
+            {
+                try {
+                    return new File(sourceFinder.findSourceFile(this).getFullFileName()).getCanonicalPath();
+                } catch (IOException e) {
+                    AnalysisContext.logError("Error resolving Real SourcePath (only relative source path will be available) ", e);
+                }
+            }
+            else {
+                AnalysisContext.logError("No SourceFinder found (only relative source path will be available) ");
+            }
+        }
+        return getSourcePath();
+    }
+
+    private SourceFinder getSourceFinder() {
```

Comment:
Please annotate this method with `@CheckReturnValue`, to tell that returned value is nullable.